Mathlib/Topology/ExtremallyDisconnected.lean

```diff
@@ -139,6 +136,132 @@ protected theorem CompactT2.Projective.extremallyDisconnected [CompactSpace X] [
+    (π_cont : Continuous π) (π_surj : π.Surjective) : ∃ E : Set D, CompactSpace E ∧ π '' E = univ ∧
+    ∀ E₀ : Set E, E₀ ≠ univ → IsClosed E₀ → E.restrict π '' E₀ ≠ univ := by
+  let S : Set <| Set D := {E : Set D | IsClosed E ∧ π '' E = univ}
+  suffices ∀ (C : Set <| Set D) (_ : C ⊆ S) (_ : IsChain (· ⊆ ·) C), ∃ s ∈ S, ∀ c ∈ C, s ⊆ c by
+    rcases zorn_superset S this with ⟨E, ⟨E_closed, E_surj⟩, E_min⟩
+    refine ⟨E, isCompact_iff_compactSpace.mp E_closed.isCompact, E_surj, ?_⟩
+    intro E₀ E₀_min E₀_closed
+    contrapose! E₀_min
+    exact eq_univ_of_coe_eq <|
+      E_min E₀ ⟨E₀_closed.trans E_closed, image_coe_eq_restrict_image ▸ E₀_min⟩ coe_subset
+  intro C C_sub C_chain
+  refine ⟨iInter (fun c : C => c), ⟨isClosed_iInter fun ⟨_, h⟩ => (C_sub h).left, ?_⟩,
+    fun c hc _ h => mem_iInter.mp h ⟨c, hc⟩⟩
+  by_cases hC : Nonempty C
+  · refine eq_univ_of_forall fun a => inter_nonempty_iff_exists_left.mp ?_
+    refine iInter_inter (ι := C) (π ⁻¹' {a}) _ ▸
+      IsCompact.nonempty_iInter_of_directed_nonempty_compact_closed _
+      ?_ (fun c => ?_) (fun c => IsClosed.isCompact ?_) (fun c => ?_)
+    · replace C_chain : IsChain (· ⊇ ·) C := C_chain.symm
+      have : ∀ s t : Set D, s ⊇ t → _ ⊇ _ := fun _ _ => inter_subset_inter_left <| π ⁻¹' {a}
+      exact (directedOn_iff_directed.mp C_chain.directedOn).mono_comp (· ⊇ ·) this
+    · rw [← image_inter_nonempty_iff, (C_sub c.mem).right, univ_inter]
+      exact singleton_nonempty a
+    all_goals exact (C_sub c.mem).left.inter <| (T1Space.t1 a).preimage π_cont
+  · rw [@iInter_of_empty _ _ <| not_nonempty_iff.mp hC, image_univ_of_surjective π_surj]
+
+/-- Gleason's Lemma 2.1: if $\rho$ is a continuous surjection from a topological space $E$ to a
+topological space $A$ satisfying the "Zorn subset condition", then $\rho(G)$ is contained in the
+closure of $A \setminus \rho(E \setminus G)}$ for any open set $G$ of $E$. -/
+lemma image_subset_closure_compl_image_compl_of_isOpen {ρ : E → A} (ρ_cont : Continuous ρ)
+    (ρ_surj : ρ.Surjective) (zorn_subset : ∀ E₀ : Set E, E₀ ≠ univ → IsClosed E₀ → ρ '' E₀ ≠ univ)
+    {G : Set E} (hG : IsOpen G) : ρ '' G ⊆ closure ((ρ '' Gᶜ)ᶜ) := by
+  by_cases G_empty : G = ∅
+  · simpa only [G_empty, image_empty] using empty_subset _
+  · intro a ha
+    rw [mem_closure_iff]
+    intro N N_open hN
+    rcases (G.mem_image ρ a).mp ha with ⟨e, he, rfl⟩
+    have nonempty : (G ∩ ρ⁻¹' N).Nonempty := ⟨e, mem_inter he <| mem_preimage.mpr hN⟩
+    have is_open : IsOpen <| G ∩ ρ⁻¹' N := hG.inter <| N_open.preimage ρ_cont
+    have ne_univ : ρ '' (G ∩ ρ⁻¹' N)ᶜ ≠ univ := zorn_subset _ (compl_ne_univ.mpr nonempty)
+                                                  is_open.isClosed_compl
+    rcases nonempty_compl.mpr ne_univ with ⟨x, hx⟩
+    have hx' : x ∈ (ρ '' Gᶜ)ᶜ := (compl_subset_compl.mpr <| image_subset ρ <|
+                                    compl_subset_compl.mpr <| G.inter_subset_left _) hx
```

Comment:
The indentation here is a bit wonky (although I realize we don't have an updated style guide), but in any case you can make `simp` do some work for you.
```suggestion
    have hx' : x ∈ (ρ '' Gᶜ)ᶜ := fun h ↦ hx <| image_subset ρ (by simp) h
```